src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,746 @@
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    case TL_ACL_DATA_PKT_TYPE: {
+        TL_AclDataSerial_t *acl = &(((TL_AclDataPacket_t *)hcievt)->AclDataSerial);
+        __disable_irq();
+        len = acl->length + 5;
+        if (len <= BLE_MODULE_SHARED_MEM_BUFFER_SIZE - _write_index) {
+          memcpy((uint8_t *)&_rxbuff[_write_index], (uint8_t *)acl, len);
+          /* move index */
+          _write_index += len;
+        } else {
+          data_overflow = true;
+        }
+        __enable_irq();
+      }
+      break;
+    default:
+      /* should not happen */
+#if defined(PRINT_IPCC_INFO)
+      printf("BLE TL evt_received, wrong type:%d\r\n", hcievt->evtserial.type);
+#endif /*(PRINT_IPCC_INFO)*/
+      while (1); /* let's block to check */
+      break;
+  }
+#if defined(PRINT_IPCC_INFO)
+  if (data_overflow) {
+    printf("Error: data read overflow\r\n");
+  }
+#endif /*(PRINT_IPCC_INFO)*/
+
+  /*  In case Event belongs to the Evt Pool we need to inform  */
+  if (((uint8_t *)hcievt >= EvtPool) && ((uint8_t *)hcievt < (EvtPool + POOL_SIZE))) {
+    /*  Free the message from shared memory */
+    TL_MM_EvtDone(hcievt);
+  }
+}
+
+/* to send BLE packet to the SharedMem */
+uint16_t mbox_write(uint8_t type, uint16_t len, const uint8_t *pData)
+{
+  TL_CmdPacket_t *bleCmdBuf = &BleCmdBuffer;
+  // Note: Until enum is available
+  // type 01 Command
+  // type 02 ACL DATA
+  // type 03 SCO Voice (not supported)
+  // type 04 event - uplink (not supported)
+
+  /*  TO DO : MANAGE ACL DATA CASE in separate buffer */
+  switch (type) {
+    case 1: { //BLE command
+        bleCmdBuf->cmdserial.type = type; // for now this param is overwritten in TL_BLE_SendCmd
+        bleCmdBuf->cmdserial.cmd.plen = len;
+        memcpy((void *) &bleCmdBuf->cmdserial.cmd, pData, len);
+        /* We're tracing here the command, after copy in shared mem but before M0 trigger. */
+        TL_BLE_SendCmd(NULL, 0); // unused parameters for now
+      }
+      break;
+    case 2: { //ACL DATA
+        if (!acl_data_wait()) {
+#if defined(PRINT_IPCC_INFO)
+          printf("ERROR: previous ACL message not ACK'd\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+          /*  return number of bytes sent, 0 in this error case */
+          return 0;
+        }
+        TL_AclDataSerial_t *aclDataSerial = (TL_AclDataSerial_t *)(HciAclDataBuffer + sizeof(TL_PacketHeader_t));
+        aclDataSerial->type = type; // for now this param is overwritten in TL_BLE_SendCmd
+        memcpy(HciAclDataBuffer + + sizeof(TL_PacketHeader_t) + sizeof(type), pData, len);
+        TL_BLE_SendAclData(NULL, 0); // unused parameters for now
+        acl_data_on = true; /* data being send */
+      }
+      break;
+    default:
+#if defined(PRINT_IPCC_INFO)
+      printf("ERROR: not supported type\r\n");
+#endif /*(PRINT_IPCC_INFO)*/
+      while (1);
```

Comment:
Again I would not block a running code, or only in debug mode, not in release mode

Reply:
will return 'no data written'